lib/rubocop/cop/rails/exit.rb

```diff
@@ -0,0 +1,50 @@
+# encoding: utf-8
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop enforces that 'exit' calls are not used within a rails app.
+      # Valid options are instead to raise an error, break, return or some
+      # other form of stopping execution of current request.
+      #
+      # There are two obvious cases where 'exit' is particularly harmful:
+      #
+      # - Usage in library code for your application. Even though rails will
+      # rescue from a SystemExit and continue on, unit testing that library
+      # code will result in specs exiting (potentially silently if exit(0)
+      # is used.)
+      # - Usage in application code outside of the web process could result in
+      # the program exiting, which could result in the code failing to run and
+      # do its job.
+      #
+      # If user still needs to call 'exit', they may do so with Process.exit
+      # or Kernel.exit
+      class Exit < Cop
+        include ConfigurableEnforcedStyle
+
+        MESSAGE = 'Do not use `exit` in Rails applications.'.freeze
+        ALLOWED_METHODS = [:exit, :exit!].freeze
+        SKIPPED_RECEIVERS = [:Kernel, :Process].freeze
+
+        def on_send(node)
+          add_offense(node, :selector) if offending_node?(node)
+        end
+
+        private
+
+        def offending_node?(node)
+          receiver_node, method_name, *arg_nodes = *node
+          _a, receiver_node_class, _c = *receiver_node
+          ALLOWED_METHODS.include?(method_name) &&
+            !SKIPPED_RECEIVERS.include?(receiver_node_class) &&
+            (arg_nodes.nil? || [0, 1].include?(arg_nodes.length))
+        end
+
+        def message(*)
```

Comment:
This method is redundant. Just pass the message explicitly to the method registering the offence.